aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -44,4 +46,11 @@ module aptos_framework::gas_schedule {
         // Need to trigger reconfiguration so validator nodes can sync on the updated gas schedule.
         reconfiguration::reconfigure();
     }
+
+    public fun set_storage_gas_config(aptos_framework: &signer, config: StorageGasConfig) {
```

Comment:
Why do we need this extra function? Can't we just call storage_gas::set_config directly?

Reply:
we can. But I think @vgao1996 insists we control gas-related params from a single module.

Reply:
> Can't we just call storage_gas::set_config directly?

I'm totally fine with calling `storage_gas::set_config` directly.

> @vgao1996 insists we control gas-related params from a single module.

Well I did prefer us having a single entity to manage all gas-related stuff, but my definition for this single entity is pretty broad -- the combined group of "gas_schedule" & "storage" gas is conceptually still a single entity to me ;)

Reply:
With that being said I remember there were some technical issues preventing us from having everything in the same module in the first place, that is, modules cannot have cyclic dependencies and I have a feeling that the same issue also applies here.

Correct me if I'm wrong:

When we update the gas schedule, we want to call `reconfigure()`, so the module this entry point is in needs to depend on `reconfiguration`. However, `storage_gas` has an `on_reconfig` event so the reconfiguration module needs to depend on it. Therefore we cannot really call `reconfigure()` in `storage_gas`, but have to do it in a different module...

Reply:
@vgao1996 is correct about the reconfig. Yes, we want `reconfig` after set_storage_gas_config but we cannot do it inside storage_gas. we can only `set_config` inside.